### PR TITLE
🐛 Fix bug with partnerName and firstName in summary

### PR DESCRIPTION
**Pull request by Sven65**
## Explain the changes you’ve made

In accordance with #CU-1vz8d6z, I've fixed a bug that made the firstName of the applicant show up as the name of the co-applicant when the co-applicant viewed the summary of a case.

## Explain your solution

I've added the persons array to the formState context and used it in `formActions#replaceMarkdownText` which is responsible for providing the text replacer used with the data.

## How to test

Concrete example:

1. Checkout this branch
2. Get a case with co-applicant
3. Follow it through, submit incomes and expenses for both the applicant and the co-applicant
4. Check the summary as a co-applicant
5. See that the names are correct

## Tested environments

- [] Storybook on a iOS device/simulator.
- [x] Building the Application on a iOS device/simulator.
- [] Building the Application on a Android device/simulator.

## Other notes

I am unsure if my solution is the best and I'm not super happy with the code. I'm unsure how to make it better and I'd like some feedback on that.

---

## Checklist for reviewers

Here are a few points that are useful for reviewers to check.

The short list of the (maybe) most important points are:

- Does this PR work as expected?
- Does the PR properly only concern a single feature/bug/thing?
- Is the code clear and concise?
- Will the changes impact existing functionality?
- Are there any linting errors (ESLint) that should be resolved?
- Do the automatic GitHub checks pass?

However it is good if a reviewer can be as thorough as possible,
within reasonable limits of course.

### Functionality

- Does this PR work as expected?
- If this is a new feature, does it add value, or is it a sign of feature-creep?
- Will the changes impact existing functionality?
- Does the changes create inconsistencies somewhere else?
- Does the code work on different platforms if applicable?
- How does the code handle if unexpected values are passed to functions etc.?

### Readability, syntax, formatting, style

- Is the code clear and concise?
- Has styling rules (Prettier) been applied to included files?
- Are functions/variables/etc. aptly named?

### Design

- Is the code properly planned and designed?
- Is code organized/separated in a logical fashion?
- Does the code follow clean code practices to a reasonable degree?
- Is the code self-documenting enough?
- Is the code in sync with existing patterns/technologies?
- Does the code reuse existing functionality where applicable?
- Is the code designed with reusability in mind?
- Does the code have hard-coded values that can be refactored?

### Patterns, idioms, tests, best practices

- Are there any linting errors (ESLint) that should be resolved?
- Is the code properly typed (TypeScript)?
- Does the code utilize language features to their full potential?

### Debugging, testing, configuring

- Is appropriate logging used?
- Have tests been added/modified to accommodate the changes?
- Is the feature sufficiently configurable in case tweaks are required in the future?

### Performance, reliability, scalability

- Is the code well optimized?
- Does the code scale?
- Does the code handle failure scenarios?

### Security

- Is there any obvious potential for exploits and security threats?
- Does code check proper authorization where applicable?

### Etiquettes

- Are commits atomic (does only one thing per commit)?
- Does the PR properly only concern a single feature/bug/thing?
- Are commit messages well-written?
- Do the automatic GitHub checks pass?

### CODE REVIEW

**Comment by helsingborgsstad** [comment]
Task linked: [CU-1vz8d6z Bugg: När medsökande granskar summeringen av ansökan blir både #firstName och #partnerName medsökandes namn](https://app.clickup.com/t/1vz8d6z)

File: source/containers/Form/hooks/textReplacement.ts [hunk 1]
```diff
@@ -6,7 +6,7 @@ import { deepCopy } from "../../../helpers/Objects";
 
 import { Step } from "../../../types/FormTypes";
 import { PartnerInfo, User } from "../../../types/UserTypes";
-import { Case } from "../../../types/Case";
+import { Case, Person } from "../../../types/Case";
```
Comment (MasterKenth) [inline source/containers/Form/hooks/textReplacement.ts:9]: `Person` import in this file seems unused?

File: source/containers/Form/hooks/formActions.ts [hunk 2]
```diff
@@ -4,6 +4,23 @@ import { FormReducerState } from "./useForm";
...
+    partnerPersonalid: person.personalNumber,
+  };
+}
+
+function computePersonUserInfo(person: Person): User {
+  return {
+    firstName: person?.firstName,
+    lastName: person?.lastName,
+  };
+}
```
Comment (MasterKenth) [inline source/containers/Form/hooks/formActions.ts:23]: This seems to only return properties with the same name, could we just `return person`? Or even remove this function altogether? Unless there's a reason to _only_ include these properties (in which case we should change the return type to `Partial<User>`).

**Review by MasterKenth — CHANGES_REQUESTED** [review]
Seems to work as intended! I just have a few minor pointers 🙂.

File: source/containers/Form/hooks/formActions.ts [hunk 3]
```diff
@@ -15,13 +32,28 @@ export function replaceMarkdownText(state: FormReducerState) {
     user,
     period,
     formAnswers: { partnerInfo },
+    persons,
   } = state;
 
+  const partnerPerson = persons.find((person) => person.role === "coApplicant");
+  const applicantPerson = persons.find((person) => person.role === "applicant");
+
+  let computedPartnerInfo: PartnerInfo = partnerPerson
```
Comment (jlarsson) [inline source/containers/Form/hooks/formActions.ts:41]: let -> const

**Review by jlarsson — COMMENTED** [review]
Consise and tidy solution

File: source/containers/Form/hooks/textReplacement.ts (hunk 1 above)
Comment (Sven65) [reply source/containers/Form/hooks/textReplacement.ts]: Correct, removed the import

File: source/containers/Form/hooks/formActions.ts (hunk 2 above)
Comment (Sven65) [reply source/containers/Form/hooks/formActions.ts]: Removed the function.

File: source/containers/Form/hooks/formActions.ts [hunk 4]
```diff
@@ -15,13 +32,28 @@ export function replaceMarkdownText(state: FormReducerState) {
...
+    : user;
+
   const updatedSteps = replaceMarkdownTextInSteps(
     steps,
-    user,
+    computedApplicant,
     period,
-    partnerInfo
+    computedPartnerInfo
   );
```
[the remark this answers is not in the document]
Comment (Sven65) [reply source/containers/Form/hooks/formActions.ts:54]: I am unsure how to add more tests to the replaceMarkdownTextInSteps function other than those we have, as it doesn't take other inputs that we haven't tested yet.

File: source/containers/Form/hooks/formActions.ts [hunk 5]
```diff
@@ -15,13 +32,28 @@ export function replaceMarkdownText(state: FormReducerState) {
     user,
     period,
     formAnswers: { partnerInfo },
+    persons,
   } = state;
 
+  const partnerPerson = persons.find((person) => person.role === "coApplicant");
+  const applicantPerson = persons.find((person) => person.role === "applicant");
```
[the remark this answers is not in the document]
Comment (Sven65) [reply source/containers/Form/hooks/formActions.ts]: Moved this to it's own testable function.

File: source/containers/Form/hooks/formActions.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (jlarsson) [reply source/containers/Form/hooks/formActions.ts:54]: I stand corrected. The tests are actually fine.

**Review by jlarsson — COMMENTED** [review]
Good!

**Review by MasterKenth — APPROVED** [review]

File: source/containers/Form/hooks/useForm.ts [hunk 6]
```diff
@@ -127,9 +132,11 @@ function useForm(initialState: FormReducerState) {
    * Function for handling a on submit action in the form.
    * @param {func} callback callback function to be called on form submit.
    */
-  const handleSubmit = (callback: (formAnswers: Record<string, any>) => void) => {
+  const handleSubmit = (
+    callback: (formAnswers: Record<string, any>) => void
```
Comment (karlknowit) [inline source/containers/Form/hooks/useForm.ts:136]: don't we have a type for this?
And also, don't use any.

File: source/containers/Form/hooks/useForm.ts [hunk 7]
```diff
@@ -151,21 +158,33 @@ function useForm(initialState: FormReducerState) {
    * Function to trigger when a form field looses focus.
    */
   const handleBlur = (answer: Record<string, any>, questionId: string) => {
```
Comment (karlknowit) [inline source/containers/Form/hooks/useForm.ts:160]: don't we have a type for answer?

File: source/containers/Form/hooks/useForm.ts [hunk 8]
```diff
@@ -151,21 +158,33 @@ function useForm(initialState: FormReducerState) {
...
+    });
   };
   /**
    * Function for updating answer.
    */
-  const handleInputChange = (answer: Record<string, any>, questionId: string) => {
-    dispatch({ type: 'UPDATE_ANSWER', payload: answer });
-    dispatch({ type: 'VALIDATE_ANSWER', payload: { answer, id: questionId, checkIfDirty: true } });
+  const handleInputChange = (
+    answer: Record<string, any>,
```
Comment (karlknowit) [inline source/containers/Form/hooks/useForm.ts:171]: same here

File: source/containers/Form/hooks/useForm.ts [hunk 9]
```diff
@@ -151,21 +158,33 @@ function useForm(initialState: FormReducerState) {
...
+  ) => {
+    dispatch({ type: "UPDATE_ANSWER", payload: answer });
+    dispatch({
+      type: "VALIDATE_ANSWER",
+      payload: { answer, id: questionId, checkIfDirty: true },
+    });
   };
-  
+
   const handleAddAnswer = (answer: Record<string, any>, questionId: string) => {
```
Comment (karlknowit) [inline source/containers/Form/hooks/useForm.ts:181]: ... and here

**Review by karlknowit — APPROVED** [review]

**Review by jlarsson — APPROVED** [review]
Good enough :)

**Pull request merged**